### PR TITLE
DOC-6257 - updates due to version changes - see ticket for details

### PR DESCRIPTION
**Pull request by paulmoloneyr3**

### CODE REVIEW

File: content/en/platform/corda/4.12/community/getting-set-up.md [hunk 1]
```diff
@@ -19,26 +19,24 @@ title: Getting set up for CorDapp development
...
 * Git
 * Gradle
 
 {{< note >}} Visit the [platform support matrix]({{< relref "release-platform-support-matrix.md" >}}) to check the latest third-party application versions supported by Corda. {{< /note >}}
 
-## Installing the Java 8 JDK
+## Installing the Java 17 JDK
 
-Install the Java 8 JDK. Corda requires at least version 8u171, but do not currently support Java 9 or higher for this version of Corda.
+Install the Java 17 JDK.
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/community/getting-set-up.md:31]: Install the Java 17 JDK. Corda requires at least version 17.0.9, but does not currently support Java 18 or higher for this version of Corda.

File: content/en/platform/corda/4.12/community/getting-set-up.md [hunk 2]
```diff
@@ -19,26 +19,24 @@ title: Getting set up for CorDapp development
...
 {{< note >}} Visit the [platform support matrix]({{< relref "release-platform-support-matrix.md" >}}) to check the latest third-party application versions supported by Corda. {{< /note >}}
 
-## Installing the Java 8 JDK
+## Installing the Java 17 JDK
 
-Install the Java 8 JDK. Corda requires at least version 8u171, but do not currently support Java 9 or higher for this version of Corda.
+Install the Java 17 JDK. 
 
-Corda has been tested against the following Java builds:
+Corda 4.12 has been tested against the following Java builds:
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/community/getting-set-up.md:33]: At what point does it become presumed that this page is talking about 4.12? im a bit wary about keeping consistency between versions, if we haven't mentioned the version before, should we start now?

(obviously up to you, just something I noticed)

File: content/en/platform/corda/4.12/community/getting-set-up.md [hunk 3]
```diff
@@ -19,26 +19,24 @@ title: Getting set up for CorDapp development
...
 
-  * [Oracle JDK](https://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
-  * [Amazon Corretto](https://aws.amazon.com/corretto/)
-  * [Red Hat’s OpenJDK](https://developers.redhat.com/products/openjdk/overview/)
-  * [Zulu’s OpenJDK](https://www.azul.com/)
+  * [Oracle JDK 17.0.9](https://www.oracle.com/java/technologies/javase/jdk17-archive-downloads.html)
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/community/getting-set-up.md:35]: again with consistency, I think mention the version above and then just talk about generic JDK builds that are compatible, the user can then find that for themselves:

> [Oracle JDK]

File: content/en/platform/corda/4.12/community/getting-set-up.md [hunk 4]
```diff
@@ -19,26 +19,24 @@ title: Getting set up for CorDapp development
...
+  * [Zulu’s OpenJDK 17.46.19 (17.0.9+8)](https://www.azul.com/downloads/?version=java-17-lts&show-old-builds=true#zulu)
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/community/getting-set-up.md:36]: Again just,

> [Zulu’s OpenJDK](https://www.azul.com/)

and keep the link as before, unless you want to guide a user more specifically

File: content/en/platform/corda/4.12/community/getting-set-up.md [hunk 5]
```diff
@@ -19,26 +19,24 @@ title: Getting set up for CorDapp development
...
 
 {{< note >}}
-OpenJDK builds often exclude JavaFX, which is required by the Corda GUI tools. Corda supports only Java 8.
+OpenJDK builds often exclude JavaFX, which is required by the Corda GUI tools.
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/community/getting-set-up.md:39]: > Corda supports only Java 17.

add this sentence back in

File: content/en/platform/corda/4.12/community/getting-set-up.md [hunk 6]
```diff
@@ -62,7 +60,7 @@ To install Git, navigate to [Git](https://git-scm.com/) and install your preferr
 
 If you intend to proceed directly to run a sample CorDapp, as described in [Running a sample CorDapp]({{< relref "tutorial-cordapp.md" >}}), the included `gradlew` script should install Gradle automatically when you open the sample CorDapp in IntelliJ.
 
-If you'd prefer to install Gradle manually, navigate to [Gradle](https://gradle.org/releases/) then locate and install Gradle **version 5.6.4**. Corda requires a Gradle version between 5.1 and 5.6.4, and does not support Gradle 6.x.
+If you'd prefer to install Gradle manually, navigate to [Gradle](https://gradle.org/releases/) then locate and install Gradle **version 7.6.4**.
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/community/getting-set-up.md:63]: > Corda requires Gradle version 7.6.x, and does not support Gradle 8.x.

add this info back

File: content/en/platform/corda/4.12/community/node-upgrade-notes.md [hunk 7]
```diff
@@ -74,7 +74,7 @@ Make sure it’s available on your path, and that you’ve read the [Release not
 node requires.
 
 {{< important >}}
-Corda 4 requires Java 8u171 or any higher Java 8 patch level. Java 9+ is not currently supported.
+Corda 4 requires Java 17.
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/community/node-upgrade-notes.md:77]: ### for this line:

> Corda 4 requires Java 17.0.9 or any higher Java 17 patch level. Java 18+ is not currently supported.

### for this file:

Please can you also add the changes to this file from this PR:
https://github.com/corda/corda-docs-portal/pull/2176/files#diff-415c0ea577a3609930e1405747ef6e3f7c2b129e6c5299345c92befef84c7290

There is a bit of re-jigging of steps

File: content/en/platform/corda/4.12/enterprise/app-upgrade-notes-enterprise.md [hunk 8]
```diff
@@ -83,7 +83,7 @@ been removed.
 
    You can find instructions for both options in [Running tests in IntelliJ]({{< relref "testing.md#running-tests-in-intellij" >}}).
 
-4. Check you're using Corda Gradle plugins version 5.0.12, and that Corda Enterprise dependencies are referenced with the right distribution.
+4. Check you're using Corda Gradle plugins version 5.0.18, and that Corda Enterprise dependencies are referenced with the right distribution.
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/enterprise/app-upgrade-notes-enterprise.md:86]: Please also add the changes to this file from this PR:

https://github.com/corda/corda-docs-portal/pull/2176/files#diff-f841687c1077226686eda2af744b4474dbd9f8eed7a748eb8ae27b6223859edc

File: content/en/platform/corda/4.12/enterprise/cordapps/getting-set-up.md [hunk 9]
```diff
@@ -16,21 +16,19 @@ weight: 10
...
 # Get set up for CorDapp development
 
-Before you start developing CorDapps, you need to download four pieces of software: the Java 8 JDK, IntelliJ IDEA, Git, and Gradle.
+Before you start developing CorDapps, you need to download four pieces of software: the Java 17 JDK, IntelliJ IDEA, Git, and Gradle.
 
-## Install the Java 8 JDK
+## Install the Java 17 JDK
 
-Install the Java 8 JDK. Corda requires at least version 8u171, but does not currently support Java 9 or higher.
+Install the Java 17 JDK.
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/enterprise/cordapps/getting-set-up.md:23]: add this back in:

>Corda requires at least version 17.0.9, but does not currently support Java 18 or higher.

File: content/en/platform/corda/4.12/enterprise/cordapps/getting-set-up.md [hunk 10]
```diff
@@ -16,21 +16,19 @@ weight: 10
...
+Before you start developing CorDapps, you need to download four pieces of software: the Java 17 JDK, IntelliJ IDEA, Git, and Gradle.
 
-## Install the Java 8 JDK
+## Install the Java 17 JDK
 
-Install the Java 8 JDK. Corda requires at least version 8u171, but does not currently support Java 9 or higher.
+Install the Java 17 JDK. 
 
-Corda has been tested against the following Java builds:
+Corda 4.12 has been tested against the following Java builds:
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/enterprise/cordapps/getting-set-up.md:25]: same comment as above

File: content/en/platform/corda/4.12/enterprise/cordapps/getting-set-up.md [hunk 11]
```diff
@@ -16,21 +16,19 @@ weight: 10
...
 
-  * [Oracle JDK](https://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
-  * [Amazon Corretto](https://aws.amazon.com/corretto/)
-  * [Red Hat’s OpenJDK](https://developers.redhat.com/products/openjdk/overview/)
-  * [Zulu’s OpenJDK](https://www.azul.com/)
+  * [Oracle JDK 17.0.9](https://www.oracle.com/java/technologies/javase/jdk17-archive-downloads.html)
+  * [Zulu’s OpenJDK 17.46.19 (17.0.9+8)](https://www.azul.com/downloads/?version=java-17-lts&show-old-builds=true#zulu)
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/enterprise/cordapps/getting-set-up.md:28]: same comment as above for oracle and zulu

File: content/en/platform/corda/4.12/enterprise/cordapps/getting-set-up.md [hunk 12]
```diff
@@ -16,21 +16,19 @@ weight: 10
...
 
 {{< note >}}
-  OpenJDK builds often exclude JavaFX, which is required by the Corda GUI tools. Corda supports only Java 8.
+  OpenJDK builds often exclude JavaFX, which is required by the Corda GUI tools.
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/enterprise/cordapps/getting-set-up.md:31]: same comment as above, add this back in:

>Corda supports only Java 17.

File: content/en/platform/corda/4.12/enterprise/cordapps/getting-set-up.md [hunk 13]
```diff
@@ -55,7 +53,7 @@ To install Git, go to [Git](https://git-scm.com/) and install the version for yo
 Gradle is a build automation tool for multi-language software development. It controls the development process in the tasks of compilation and packaging to testing, deployment, and publishing.
 
 * If you [run the sample CorDapp]({{< relref "tutorial-cordapp.md" >}}), the included `gradlew` script installs Gradle automatically when you open the sample CorDapp in IntelliJ.
-* If you plan to start developing your own CorDapp straight away, install Gradle manually. Go to [Gradle](https://gradle.org/releases/), then locate and install Gradle **version 5.6.4**. Corda requires a Gradle version between 5.1 and 5.6.4, and does not support Gradle 6.x.
+* If you plan to start developing your own CorDapp straight away, install Gradle manually. Go to [Gradle](https://gradle.org/releases/), then locate and install Gradle **version 7.6.4**.
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/enterprise/cordapps/getting-set-up.md:56]: add this back in (same as above):

> Corda requires Gradle version 7.6.x, and does not support Gradle 8.x.

File: content/en/platform/corda/4.12/enterprise/minor-version-node-upgrade.md [hunk 14]
```diff
@@ -50,7 +50,7 @@ Download the latest version of Corda from the [Customer Hub](https://customerhub
 Make sure it’s available on your path, and that you’ve read the [Corda release notes]({{< relref "release-notes-enterprise.md" >}}). Pay particular attention to which version of Java the node requires.
 
 {{< important >}}
-Corda 4 requires Java 8u171 or any higher Java 8 patch level. Java 9+ is not currently supported.
+Corda 4 requires Java 17.
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/enterprise/minor-version-node-upgrade.md:53]: > Corda 4 requires Java 17.0.9 or any higher Java 17 patch level. Java 18+ is not currently supported.

File: content/en/platform/corda/4.12/enterprise/node-upgrade-notes.md [hunk 15]
```diff
@@ -306,7 +306,7 @@ node requires.
 
 
 {{< important >}}
-Corda 4 requires Java 8u171 or any higher Java 8 patch level. Java 9+ is not currently supported.
+Corda 4 requires Java 17.
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/enterprise/node-upgrade-notes.md:309]: ### for this line:

> Corda 4 requires Java 17.0.9 or any higher Java 17 patch level. Java 18+ is not currently supported.

### for this file:

Please can you also add the changes to this file from this PR:
https://github.com/corda/corda-docs-portal/pull/2176/files#diff-c7f6e94ae2a65c463894d51e4b7675b67328f83cb40e252fb67969143f640448


There are quite a lot of changes to this file needed

File: content/en/platform/corda/4.12/enterprise/operations/project-planner/cordapp-developer.md [hunk 16]
```diff
@@ -22,8 +22,8 @@ When planning to develop a CorDapp, consider the following factors:
 
 There are a number of pre-requisites for CorDapp development.
 
-- You must use the **Java 8 JVM**, version 8u171 and onwards are supported, but version 9 and later is not supported.
-- Gradle 5.4.1
+- You must use Java 17.
```
Comment (tomstark99) [inline content/en/platform/corda/4.12/enterprise/operations/project-planner/cordapp-developer.md:25]: > You must use the **Java 17 JVM**, version 17.0.9 and onwards are supported, but version 18 and later is not supported.

**Review by tomstark99 — DISMISSED** [review]
please could you also remove this line:

https://github.com/corda/corda-docs-portal/pull/2176/files#diff-f2068a2c0d6c7f05f4f753bc6f2695ea0c909d0101d60c9106160ab3ccd0fba8

from `release-checksum-enterprise.md`

(sorry quite a lot of changes for this one)

File: content/en/platform/corda/4.12/community/getting-set-up.md (hunk 2 above)
Comment (paulmoloneyr3) [reply content/en/platform/corda/4.12/community/getting-set-up.md:33]: I thought it read better with the version but will remove for consistency.

File: content/en/platform/corda/4.12/community/building-against-non-release.md [hunk 17]
```diff
@@ -63,4 +63,4 @@ time may differ. If you are using an unstable release and need help debugging an
 * Make a note of the `corda_release_version` in the root `build.gradle` file of the Corda repository.
 * In your CorDapp’s root `build.gradle` file:
   * Update `ext.corda_release_version` to the `corda_release_version` noted down earlier.
-  * Update `corda_gradle_plugins_version` to the `gradlePluginsVersion` noted down earlier - for Corda Enterprise Edition 4.12 this must be `ext.corda_gradle_plugins_version = '5.0.12'`.
+  * Update `corda_gradle_plugins_version` to the `gradlePluginsVersion` noted down earlier - for Corda Enterprise Edition 4.12 this must be `ext.corda_gradle_plugins_version = '5.0.18'`.
```
Comment (adelel1) [inline content/en/platform/corda/4.12/community/building-against-non-release.md:66]: should be ext.corda_gradle_plugins_version = '5.1.1'

File: content/en/platform/corda/4.12/community/cordapp-build-systems.md [hunk 18]
```diff
@@ -80,10 +80,10 @@ ext.corda_release_distribution = 'com.r3.corda'
 ext.corda_core_release_distribution = 'net.corda'
 ext.corda_release_version = '4.12'
 ext.corda_core_release_version = '4.12'
-ext.corda_gradle_plugins_version = '5.0.12'
-ext.quasar_version = '0.7.15_r3'
+ext.corda_gradle_plugins_version = '5.0.18'
```
Comment (adelel1) [inline content/en/platform/corda/4.12/community/cordapp-build-systems.md:83]: 5.1.1

File: content/en/platform/corda/4.12/enterprise/app-upgrade-notes-enterprise.md [hunk 19]
```diff
@@ -41,9 +41,9 @@ To re-compile your CorDapp for Corda Enterprise Edition 4.12, you need to:
     corda_core_release_distribution = 'net.corda'
     corda_release_version = '4.12'
     corda_core_release_version = '4.12'
-    corda_gradle_plugins_version = '5.0.12'
-    kotlin_version = '1.2.71'
-    quasar_version = '0.7.16_r3'
+    corda_gradle_plugins_version = '5.0.18'
```
Comment (adelel1) [inline content/en/platform/corda/4.12/enterprise/app-upgrade-notes-enterprise.md:44]: 5.1.1

File: content/en/platform/corda/4.12/enterprise/app-upgrade-notes-enterprise.md (hunk 8 above)
Comment (adelel1) [inline content/en/platform/corda/4.12/enterprise/app-upgrade-notes-enterprise.md:86]: 5.1.1

File: content/en/platform/corda/4.12/enterprise/building-against-non-release.md [hunk 20]
```diff
@@ -57,4 +57,4 @@ time may differ. If you are using an unstable release and need help debugging an
 9. Make a note of the `corda_release_version` in the root `build.gradle` file of the Corda repository.
 10. In your CorDapp’s root `build.gradle` file:
   1. Update `ext.corda_release_version` to the `corda_release_version` you noted earlier.
-  2. Update `corda_gradle_plugins_version` to the `gradlePluginsVersion` you noted earlier - for Corda Enterprise Edition 4.12, this must be `ext.corda_gradle_plugins_version = '5.0.12'`.
+  2. Update `corda_gradle_plugins_version` to the `gradlePluginsVersion` you noted earlier - for Corda Enterprise Edition 4.12, this must be `ext.corda_gradle_plugins_version = '5.0.18'`.
```
Comment (adelel1) [inline content/en/platform/corda/4.12/enterprise/building-against-non-release.md:60]: 5.1.1

File: content/en/platform/corda/4.12/enterprise/cordapps/cordapp-build-systems.md [hunk 21]
```diff
@@ -74,10 +74,10 @@ ext.corda_release_distribution = 'com.r3.corda'
 ext.corda_core_release_distribution = 'net.corda'
 ext.corda_release_version = '4.11'
 ext.corda_core_release_version = '4.11'
-ext.corda_gradle_plugins_version = '5.0.12'
-ext.quasar_version = '0.7.15_r3'
+ext.corda_gradle_plugins_version = '5.0.18'
```
Comment (adelel1) [inline content/en/platform/corda/4.12/enterprise/cordapps/cordapp-build-systems.md:77]: 5.1.1

**Review by adelel1 — APPROVED** [review]

**Review by AnnaK-R3 — APPROVED** [review]

**Pull request merged**